daemons/gptp/common/avbts_message.hpp

```diff
@@ -223,21 +226,50 @@ class PTPMessageCommon {
 
 #pragma pack(push,1)
 
+#define PATH_TRACE_TLV_TYPE 0x8
+
 class PathTraceTLV {
  private:
 	uint16_t tlvType;
-	uint16_t lengthField;
-	ClockIdentity identity;
+	typedef std::list<ClockIdentity> IdentityList;
+	IdentityList identityList;
  public:
-	 PathTraceTLV() {
-		tlvType = PLAT_htons(0x8);
-		lengthField = PLAT_htons(sizeof(identity));
-	}
-	void setClockIdentity(ClockIdentity * id) {
-		identity = *id;
+	PathTraceTLV() {
+		tlvType = PLAT_htons(PATH_TRACE_TLV_TYPE);
+	}
+	void parseClockIdentity(uint8_t *buffer) {
+		int length = PLAT_ntohs(*((uint16_t *)buffer))/PTP_CLOCK_IDENTITY_LENGTH;
+		buffer += sizeof(uint16_t);
+		for(; length > 0; --length) {
+			ClockIdentity add;
+			add.set(buffer);
+			identityList.push_back(add);
+			buffer += PTP_CLOCK_IDENTITY_LENGTH;
+		}
+	}
+	void appendClockIdentity(ClockIdentity * id) {
+		identityList.push_back(*id);
 	}
 	void toByteString(uint8_t * byte_str) {
-		memcpy(byte_str, this, sizeof(*this));
+		IdentityList::iterator iter;
+		*((uint16_t *)byte_str) = tlvType;
+		byte_str += sizeof(tlvType);
+		*((uint16_t *)byte_str) = PLAT_htons(identityList.size()*PTP_CLOCK_IDENTITY_LENGTH);
+		byte_str += sizeof(uint16_t);
+		for
+			(iter = identityList.begin();
+			 iter != identityList.end(); ++iter) {
+			iter->getIdentityString(byte_str);
+			byte_str += PTP_CLOCK_IDENTITY_LENGTH;
+		}
+	}
+	bool has(ClockIdentity *id) {
+		return std::find
+			(identityList.begin(), identityList.end(), *id) !=
+			identityList.end();
+	}
+	int length() {
+		return 2*sizeof(uint16_t) + PTP_CLOCK_IDENTITY_LENGTH*identityList.size();
```

Comment:
comment on magic 2* would be good. I'm guessing sizeof(tlvType) + sizeof(length)?